### ISSUE TITLE
Signature extraction failed

### ISSUE THREAD

**Issue by AnkS4** [rg3/youtube-dl#12102]
> $ youtube-dl https://www.youtube.com/watch?v=gr0QS0R0uzU -F -v
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['https://www.youtube.com/watch?v=gr0QS0R0uzU', '-F', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.01.24
[debug] Python version 3.6.0 - Linux-4.9.6-1-MANJARO-x86_64-with-glibc2.3.4
[debug] exe versions: ffmpeg 3.2.2, ffprobe 3.2.2, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] gr0QS0R0uzU: Downloading webpage
[youtube] gr0QS0R0uzU: Downloading video info webpage
[youtube] gr0QS0R0uzU: Extracting video information
[youtube] {43} signature length 42.40, html5 player en_US-vfl8LqiZp
[youtube] gr0QS0R0uzU: Downloading player /yts/jsbin/player-en_US-vfl8LqiZp/base.js
ERROR: Signature extraction failed: Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1035, in _decrypt_signature
    video_id, player_url, s
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 949, in _extract_signature_function
    errnote='Download of %s failed' % player_url)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 519, in _download_webpage
    res = self._download_webpage_handle(url_or_request, video_id, note, errnote, fatal, encoding=encoding, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 426, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 406, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/lib/python3.6/site-packages/youtube_dl/YoutubeDL.py", line 2000, in urlopen
    req = sanitized_Request(req)
  File "/usr/lib/python3.6/site-packages/youtube_dl/utils.py", line 525, in sanitized_Request
    return compat_urllib_request.Request(sanitize_url(url), *args, **kwargs)
  File "/usr/lib/python3.6/urllib/request.py", line 329, in __init__
    self.full_url = url
  File "/usr/lib/python3.6/urllib/request.py", line 355, in full_url
    self._parse()
  File "/usr/lib/python3.6/urllib/request.py", line 384, in _parse
    raise ValueError("unknown url type: %r" % self.full_url)
ValueError: unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'
 (caused by ValueError("unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'",)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1035, in _decrypt_signature
    video_id, player_url, s
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 949, in _extract_signature_function
    errnote='Download of %s failed' % player_url)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 519, in _download_webpage
    res = self._download_webpage_handle(url_or_request, video_id, note, errnote, fatal, encoding=encoding, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 426, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 406, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/lib/python3.6/site-packages/youtube_dl/YoutubeDL.py", line 2000, in urlopen
    req = sanitized_Request(req)
  File "/usr/lib/python3.6/site-packages/youtube_dl/utils.py", line 525, in sanitized_Request
    return compat_urllib_request.Request(sanitize_url(url), *args, **kwargs)
  File "/usr/lib/python3.6/urllib/request.py", line 329, in __init__
    self.full_url = url
  File "/usr/lib/python3.6/urllib/request.py", line 355, in full_url
    self._parse()
  File "/usr/lib/python3.6/urllib/request.py", line 384, in _parse
    raise ValueError("unknown url type: %r" % self.full_url)
ValueError: unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1035, in _decrypt_signature
    video_id, player_url, s
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 949, in _extract_signature_function
    errnote='Download of %s failed' % player_url)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 519, in _download_webpage
    res = self._download_webpage_handle(url_or_request, video_id, note, errnote, fatal, encoding=encoding, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 426, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 406, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/lib/python3.6/site-packages/youtube_dl/YoutubeDL.py", line 2000, in urlopen
    req = sanitized_Request(req)
  File "/usr/lib/python3.6/site-packages/youtube_dl/utils.py", line 525, in sanitized_Request
    return compat_urllib_request.Request(sanitize_url(url), *args, **kwargs)
  File "/usr/lib/python3.6/urllib/request.py", line 329, in __init__
    self.full_url = url
  File "/usr/lib/python3.6/urllib/request.py", line 355, in full_url
    self._parse()
  File "/usr/lib/python3.6/urllib/request.py", line 384, in _parse
    raise ValueError("unknown url type: %r" % self.full_url)
ValueError: unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/youtube_dl/YoutubeDL.py", line 694, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 359, in extract
    return self._real_extract(url)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1624, in _real_extract
    encrypted_sig, video_id, player_url, age_gate)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1045, in _decrypt_signature
    'Signature extraction failed: ' + tb, cause=e)
youtube_dl.utils.ExtractorError: Signature extraction failed: Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1035, in _decrypt_signature
    video_id, player_url, s
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 949, in _extract_signature_function
    errnote='Download of %s failed' % player_url)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 519, in _download_webpage
    res = self._download_webpage_handle(url_or_request, video_id, note, errnote, fatal, encoding=encoding, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 426, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 406, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/lib/python3.6/site-packages/youtube_dl/YoutubeDL.py", line 2000, in urlopen
    req = sanitized_Request(req)
  File "/usr/lib/python3.6/site-packages/youtube_dl/utils.py", line 525, in sanitized_Request
    return compat_urllib_request.Request(sanitize_url(url), *args, **kwargs)
  File "/usr/lib/python3.6/urllib/request.py", line 329, in __init__
    self.full_url = url
  File "/usr/lib/python3.6/urllib/request.py", line 355, in full_url
    self._parse()
  File "/usr/lib/python3.6/urllib/request.py", line 384, in _parse
    raise ValueError("unknown url type: %r" % self.full_url)
ValueError: unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'
 (caused by ValueError("unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'",)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```
> $ youtube-dl --version
```
2017.01.24
```

> $ youtube-dl -U
```
It looks like you installed youtube-dl with a package manager, pip, setup.py or a tarball. Please use that to update.
```

> $ sudo pacman -S youtube-dl
```
warning: youtube-dl-2017.01.24-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) youtube-dl-2017.01.24-1

Total Installed Size:  10.15 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] n
```

**Comment by dstftw** [comment]
You are using an outdated version of youtube-dl. Read [this FAQ entry](https://yt-dl.org/update) if you have troubles updating.

**Comment by Starsam80** [comment]
Try `sudo pacman -Sy youtube-dl`, because it is indeed updated on the servers: https://www.archlinux.org/packages/community/any/youtube-dl/

**Comment by AnkS4** [comment]
> $ sudo pacman -Sy youtube-dl
```:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
warning: youtube-dl-2017.01.24-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) youtube-dl-2017.01.24-1

Total Installed Size:  10.15 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] n
```

**Comment by AnkS4** [comment]
Manjaro servers have 2017.01.24 as latest.

Is the issue solved in 2017.02.04?

**Comment by spaceone** [comment]
Version: 2017.01.10

Command: `/usr/bin/youtube-dl -i https://www.youtube.com/watch?v=bpXTAuHIGy8&list=PLD8FDD6314ADF5B17`

Output:
```
ERROR: Signature extraction failed: Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1011, in _decrypt_signature
    video_id, player_url, s
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 925, in _extract_signature_function
    errnote='Download of %s failed' % player_url)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 519, in _download_webpage
    res = self._download_webpage_handle(url_or_request, video_id, note, errnote, fatal, encoding=encoding, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 426, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query)
  File "/usr/lib/python3.6/site-packages/youtube_dl/extractor/common.py", line 406, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/lib/python3.6/site-packages/youtube_dl/YoutubeDL.py", line 2000, in urlopen
    req = sanitized_Request(req)
  File "/usr/lib/python3.6/site-packages/youtube_dl/utils.py", line 518, in sanitized_Request
    return compat_urllib_request.Request(sanitize_url(url), *args, **kwargs)
  File "/usr/lib/python3.6/urllib/request.py", line 329, in __init__
    self.full_url = url
  File "/usr/lib/python3.6/urllib/request.py", line 355, in full_url
    self._parse()
  File "/usr/lib/python3.6/urllib/request.py", line 384, in _parse
    raise ValueError("unknown url type: %r" % self.full_url)
ValueError: unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'
 (caused by ValueError("unknown url type: '/yts/jsbin/player-en_US-vfl8LqiZp/base.js'",)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

It's your software, I don't care if you set this to WONTFIX because my version is 1 month old.

**Comment by dstftw** [comment]
This is already fixed in 2017.01.31. Bother to read my post.
